test/extensions/filters/network/http_connection_manager/config_test.cc

```diff
@@ -2596,6 +2598,43 @@ class TestHeaderValidatorFactoryConfig : public Http::HeaderValidatorFactoryConf
   }
 };
 
+// Override the default config factory for the proto that is expected to be created when
+// header validator config was not specified.
```

Comment:
wait, why is the default factory not the default?

Reply:
This override is for the test to see what UHV config proto is created by HCM factory by default.